### PR TITLE
Compile against modular JSOO

### PR DESCRIPTION
**Pull request by vasilisp**
The transition is very smooth. We just need to depend on the new OPAM and Findlib packages and replace `XmlHttpRequest` by `Lwt_xmlHttpRequest`.

CC @hhugo

### CODE REVIEW

**Comment by hhugo** [comment]
:+1: 
It would be nice to start using the `Js_of_ocaml` namespace.

**Comment by vasilisp** [comment]
@balat @vouillon we can merge this when you are ready. There are respective OT&OS branches, and the changes on your side should be simple.

**Comment by hhugo** [comment]
I'm planning to merge https://github.com/ocsigen/js_of_ocaml/pull/575.
Can you check if it breaks eliom

File: src/lib/eliom_content_.client.ml [hunk 1]
```diff
@@ -574,7 +574,7 @@ module Html = struct
       let onreturn elt f =
         let f ev =
           let key = ev##.keyCode in
-          if key = Keycode.return then f ev;
+          if key = 13 then f ev;
```
Comment (hhugo) [inline src/lib/eliom_content_.client.ml]: `Dom_html.Keyboard_code.of_event ev = Dom_html.Keyboard_code.Enter`

**Comment by vasilisp** [comment]
It compiles now, thanks for the heads up! I probably got the `~form_args` stuff wrong (currently almost ignored by `Eliom_request.send`). I will revisit it tomorrow to hopefully remove `~form_args` from Eliom. I can't say I understand what it was for.

File: pkg/META [hunk 2]
```diff
@@ -56,7 +57,6 @@ package "client" (
   archive(byte) = "client.cma eliom_client_main.cmo"
 
   linkopts(javascript) = "+eliom.client/eliom_client.js"
-  linkopts(javascript) += "+js_of_ocaml/classlist.js"
```
Comment (hhugo) [inline pkg/META]: +js_of_ocaml-compiler/classlist.js

File: opam [hunk 3]
```diff
@@ -13,7 +13,11 @@ depends: [
   "deriving" {>= "0.6"}
   "ppx_deriving"
   "ppx_tools" {>= "0.99.3"}
-  "js_of_ocaml" {>= "2.8.2"}
+  "js_of_ocaml" {>= "dev"}
```
Comment (vouillon) [inline opam]: That should probably be `>= "3.0"`

**Pull request merged**